AsyncDisplayKit/Private/ASBaselineStackPositionedLayout.mm

```diff
@@ -0,0 +1,118 @@
+    return ((id<ASBaselineStackLayoutable>)a.layoutableObject).ascender < ((id<ASBaselineStackLayoutable>)b.layoutableObject).ascender;
+  });
+  const CGFloat maxAscender = baselineIt == positionedLayout.sublayouts.end() ? 0 : ((id<ASBaselineStackLayoutable>)(*ascenderIt).layoutableObject).ascender;
+  
+  CGPoint p = CGPointZero;
+  BOOL first = YES;
+  auto stackedChildren = AS::map(positionedLayout.sublayouts, [&](ASLayout *l) -> ASLayout *{
+    __weak id<ASBaselineStackLayoutable> textChild = (id<ASBaselineStackLayoutable>) l.layoutableObject;
+    p = p + directionPoint(stackStyle.direction, textChild.spacingBefore, 0);
+    if (first) {
+      // if this is the first item use the previously computed start point
+      p = l.position;
+    } else {
+      // otherwise add the stack spacing
+      p = p + directionPoint(stackStyle.direction, stackStyle.spacing, 0);
+    }
+    first = NO;
+    
+    // add the baseline offset. baselineOffset is only valid in the horizontal direction, so we always add to y
+    l.position = p + CGPointMake(0, baselineOffset(textStyle, l, maxAscender, maxBaseline));
+    
+    // If we are a vertical stack, add the item's descender (it is negative) to the spacing after. This will alter the stack spacing to be on baselines instead of bounding boxes
+    CGFloat spacingAfterBaseline = (stackStyle.direction == ASStackLayoutDirectionVertical) ? textChild.descender : 0;
+    p = p + directionPoint(stackStyle.direction, stackDimension(stackStyle.direction, l.size) + textChild.spacingAfter + spacingAfterBaseline, 0);
+    
+    return l;
+  });
+  
+  // The cross dimension is the max of the childrens' cross dimensions (clamped to our constraint below).
+  const auto it = std::max_element(stackedChildren.begin(), stackedChildren.end(),
+                                   [&](ASLayout *a, ASLayout *b) {
+                                     return maxDimensionForLayout(a, stackStyle) < maxDimensionForLayout(b, stackStyle);
+                                   });
+  const auto largestChildCrossSize = it == stackedChildren.end() ? 0 : maxDimensionForLayout(*it, stackStyle);
+  const auto minCrossSize = crossDimension(stackStyle.direction, constrainedSize.min);
+  const auto maxCrossSize = crossDimension(stackStyle.direction, constrainedSize.max);
+  const CGFloat crossSize = MIN(MAX(minCrossSize, largestChildCrossSize), maxCrossSize);
+  
+  // find the child with the largest height. Use that child's descender as the descender to pass back to the ASBaselineStackLayoutSpec.
+  const auto descenderIt = std::max_element(stackedChildren.begin(), stackedChildren.end(), [&](const ASLayout *a, const ASLayout *b){
+    return  a.position.y + a.size.height <  b.position.y + b.size.height;
+  });
+  const CGFloat minDescender = descenderIt == stackedChildren.end() ? 0 : ((id<ASBaselineStackLayoutable>)(*descenderIt).layoutableObject).descender;
+
+  return {stackedChildren, crossSize, maxAscender, minDescender};
+}
```

Comment:
Not gonna pretend that I fully understand the logic here, but it looks like you managed to reuse ASStackPositionedLayout and compute new positions and size based on its result. That's really smart! :+1: